modules/core/src/main/scala/com.snowplowanalytics.snowplow.databricks/processing/DatabricksUploader.scala

```diff
@@ -123,6 +124,21 @@ object DatabricksUploader {
     case nf: NotFound =>
       // NotFound exception messages are clean, short and helpful
       s"Not Found: ${nf.getMessage}"
+    case e: UnknownHostException =>
+      if (e.getMessage.matches(""".*\bnodename nor servname provided, or not known\b.*"""))
+        "Invalid Databricks hostname"
+      else
+        e.getMessage
+    case e: NullPointerException =>
+      if (e.getMessage.matches(""".*\bCannot invoke ".*getTokenEndpoint\(\)" because "jsonResponse" is null\b.*"""))
+        "Invalid Databricks hostname"
+      else
+        e.getMessage
```

Comment:
Here, you would send an alert to the customer if we get any `NullPointerException` even if it doesn't match the expected pattern.  This doesn't feel right, because we should only send an alert if we are 100% sure it is a customer-caused setup error.  There might be other scenarios where we get `NullPointerException` at runtime, but it's for some completely unrelated issue.

Could you move the `if` statement onto the same line as the `case` statement, i.e.

```scala
case e: NullPointerException if e.getMessage.matches(whatever) =>
```